DigitalLearningSolutions.Data/Services/CourseService.cs

```diff
@@ -177,6 +179,11 @@ RemovalMethod removalMethod
             return true;
         }
 
+        public int GetCourseCategoryId(int customisationId)
+        {
+            return courseDataService.GetCourseValidationDetails(customisationId).courseCategoryId!.Value;

```

Comment:
1. Agreed that it isn't a great name, but the other usage is in VerifyAdminUserCanAccessCourse so I can see why it's happened. There's not real issue renaming it to something better.
2. I guess technically no because the service could be called from anywhere, but there is a ServiceFilter on the Actions that currently call it. I never know whether we should explicitly handle nulls in cases like this, or whether we should just let the natural exception occur if the method is used unexpectedly on an ID that doesn't exist.

Reply:
Ok fair enough.
For point 2, I know what you mean. Given the wider context and the check in the servceiFilter this I imagine is fine. 
But for other usage of this method without any checks beforehand this will interupt the user journey in maybe a less controlled/graceful way. You would ideally want to be sending out a 404 status code rather than a 500 at least.

Reply:
Renamed the data service method anyway.

Reply:
Nice one!

Reply:
@AlexJacksonDS Why not just make the return type nullable and let the caller deal with the null case? Throwing if the method can't do what it promises also makes sense, but seeing ! in code makes me think, "The dev who wrote this must have been confident this couldn't return null - why is that?" So I've a slight preference for making it nullable. It would also make it consistent with the method it's calling.